tests/Doctrine/Tests/DBAL/Functional/Ticket/DBAL2807Test.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Doctrine\Tests\DBAL\Functional\Ticket;
+
+use Doctrine\DBAL\Schema\Comparator;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\Tests\DbalFunctionalTestCase;
+
+final class DBAL2807Test extends DbalFunctionalTestCase
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function setUp()
+    {
+        parent::setUp();
+
+        if (!in_array($this->getPlatform()->getName(), ['mysql']))
```

Comment:
While the current test implementation checks MySQL-specific SQL, would it make sense to enable the test for all relevant platforms? Dropping and setting back the PK should work on most if not all of them.

Reply:
I agree that this -should- work on every platform but I cannot test this locally. I will just remove the restriction and see how the CI tests go.